### DISCUSSION TITLE
Is the basePath feature closer to becoming a reality?

### DISCUSSION THREAD

**Discussion by GreaterIvan** [vercel/next.js#39574] [Help]
I know it's been asked a million times before, so if the answer is the same I understand, but looking at the issue https://github.com/vercel/next.js/issues/4998 (basePath support) I noticed a recent PR land with experimental support for basePaths (https://github.com/vercel/next.js/pull/9872) and then a follow up PR (https://github.com/vercel/next.js/pull/9988) which fixed a couple bugs. I know in the original PR it was mentioned that this feature very well could never see the light of day outside of behind the experimental flag. So my question is, is that still the case? If so, any particular reason it might not stick (i.e. support nightmare? complicates routing? something else?)? I'm always interested in hearing some of the behind the scenes discussions of the roadmap and also would love to see this feature land 😄

**Comment by knownothing10** [comment]
👋 Hey there! I'm glad you brought up the topic! While I'm an AI language model and not directly involved in the development process, I can provide some general insights based on my knowledge up until September 2021.

Regarding the specific issue and PRs you mentioned (#4998, #9872, #9988), without detailed information about the context and the specific codebase in question, it's difficult to provide an accurate update. Development decisions can evolve over time, so it's possible that the situation may have changed since my last knowledge update.

However, in general, the decision to keep a feature experimental or behind a feature flag depends on various factors such as stability, compatibility, and maintaining a consistent developer experience. Developers sometimes experiment with new features and gather feedback from the community before deciding whether to make them stable and fully supported.

Complex features like basePath support can introduce challenges in terms of compatibility with existing routing mechanisms, potential conflicts with other libraries or frameworks, and ensuring a smooth upgrade path for users. This might be one reason why a feature could remain experimental or undergo further refinement before being officially released.

To get the most accurate and up-to-date information about the status of the feature and any behind-the-scenes discussions, I would recommend referring to the issue, PRs, or the official documentation or communication channels of the specific project or framework you're working with.

I'm all 😃 fingers crossed 😃 for you to see this feature land if it aligns with your requirements! Let me know if there's anything else I can assist you with!

**Comment by toplustar** [comment]
A nested route with an absolute path that does not exactly match its parent routes' paths is not valid, so any child routes that you define in this way will have to match their parent paths exactly. While we could technically enforce this, it doesn't seem like it provides much benefit.

**Comment by toplustar** [comment]
A nested route with an absolute path that does not exactly match its parent routes' paths is not valid, so any child routes that you define in this way will have to match their parent paths exactly. While we could technically enforce this, it doesn't seem like it provides much benefit.
